### PR TITLE
Issue/1888 add dedicated doc page for config requirements

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

Part of https://github.com/inmanta/irt/issues/1888

- Add the `configuration_requirements.rst` file that reads json data from the `compatibility.json` generated during irt build-docs.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
~~- [ ] Type annotations are present~~
- [x] Code is clear and sufficiently documented
~~- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)~~
~~- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)~~
- [x] Correct, in line with design
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~
~~- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)~~

### CODE REVIEW

File: docs/configuration_requirements.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Configuration requirements
```
Comment (sanderr) [inline docs/configuration_requirements.rst:1]: I find the term "configuration requirements" not very intuitive (same for the variable name in irt and the occurence in the text below). I do have to confess I'm having trouble coming up with a good title. "Compatible versions" or "Version constraints" are the best I can think of right now, but I don't really like either of them very much. Will give this some more thought.

File: docs/configuration_requirements.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    These constraints are defined for the whole lifetime of this major version i.e.
+    they are not pinned to a specific three-digit version.
+
+
+    .. datatemplate:json:: required_configuration.json
+       :template: iso_config_requirements.tmpl
+
+.. only:: oss
+
+    This page shows the configuration requirements for version |version_major| of Inmanta.
```
Comment (sanderr) [inline docs/configuration_requirements.rst:16]: For OSS the major digit doesn't really have a special semantic meaning. I think we should use the full version here.

File: docs/configuration_requirements.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    .. datatemplate:json:: required_configuration.json
+       :template: oss_config_requirements.tmpl
```
Comment (sanderr) [inline docs/configuration_requirements.rst:19]: Why not split it in two templates? One for the "system requirements" (Python, RHEL, ...) and one for the components? Then we can put the latter in a `.. only:: iso` block but the former would be identical.

File: docs/configuration_requirements.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+**************************
+
+.. only:: iso
+
+    This page shows the configuration requirements for the Inmanta Service Orchestrator |version_major| version.
+    These constraints are defined for the whole lifetime of this major version i.e.
+    they are not pinned to a specific three-digit version.
+
+
+    .. datatemplate:json:: required_configuration.json
```
Comment (sanderr) [inline docs/configuration_requirements.rst:11]: What if this file does not exist? We probably need an empty one in this repo so we can still build the docs stand-alone?

File: docs/configuration_requirements.rst [hunk 5]
```diff
@@ -0,0 +1,19 @@
+Configuration requirements
+**************************
+
+.. only:: iso
+
+    This page shows the configuration requirements for the Inmanta Service Orchestrator |version_major| version.
+    These constraints are defined for the whole lifetime of this major version i.e.
+    they are not pinned to a specific three-digit version.
```
Comment (sanderr) [inline docs/configuration_requirements.rst:8]: We should mention somewhere that there is a machine-consumable version of these tables at \<insert link>. I think we may even want to link to both the json and the plain requirements file.

**Review by sanderr — CHANGES_REQUESTED** [review]

File: docs/configuration_requirements.rst (hunk 1 above)
Comment (Hugo-Inmanta) [reply docs/configuration_requirements.rst:1]: What about:
Title : "Compatibility"
In text: "Compatible versions"

Comment (sanderr) [reply docs/configuration_requirements.rst]: Sounds OK to me. Might be a good idea to quickly demo this next week. If anyone doesn't like the section name we'll hear it then.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: docs/conf.py
[the remark this answers is not in the document]
Comment (Hugo-Inmanta) [reply docs/conf.py:1]: Woops, yes we do. I'll add them back

File: docs/_templates/system_requirements.tmpl [hunk 6]
```diff
@@ -0,0 +1,25 @@
+.. -*- mode: rst -*-
+
+
+System requirements
+~~~~~~~~~~~~~~~~~~~
+
+
+{% if data['system_requirements']%}
+.. list-table:: System requirements
+   :widths: 25 25
```
[the remark this answers is not in the document]
Comment (Hugo-Inmanta) [reply docs/_templates/system_requirements.tmpl:10]: > What happens if you leave sphinx to do this based on the content of the table?

Works fine and looks better, I'll remove these hard coded values :+1:

File: docs/configuration_requirements.rst [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    they are not pinned to a specific three-digit version. Upper bounds may be added when a new major version is released.
+
+
+.. only:: oss
+
+    This page shows the compatibility of Inmanta version |release| with other components.
+
+
+.. datatemplate:json:: compatibility.json
+   :template: system_requirements.tmpl
```
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply docs/configuration_requirements.rst:17]: I see.

File: docs/_templates/components_requirements.tmpl [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+   {% for key, value in data["component_requirements"] | dictsort %}
+   * - {{ key|e }}
+     - {{ value|e }}
+   {% endfor %}
+
+.. note::
+
+    This information is also available as a constraints file in `requirements.components.txt <./requirements.components.txt>`_.
+    and in a machine-consumable format in the `compatibility.json <./compatibility.json>`_ file.
```
Comment (arnaudsjs) [inline docs/_templates/components_requirements.tmpl:23]:
```suggestion
    This information is also available as a constraints file in `requirements.components.txt <./requirements.components.txt>`_
    and in a machine-consumable format in the `compatibility.json <./compatibility.json>`_ file.
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: docs/_templates/components_requirements.tmpl (hunk 8 above)
Comment (arnaudsjs) [inline docs/_templates/components_requirements.tmpl:23]: A requirements file is also a machine-consumable format. But just for a specific tool, namely pip. 
```suggestion
    and in json format in the `compatibility.json <./compatibility.json>`_ file.
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**